packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -795,8 +795,14 @@ class _NestedScrollCoordinator implements ScrollActivityDelegate, ScrollHoldCont
 
   bool get hasScrolledBody {
     for (final _NestedScrollPosition position in _innerPositions) {
-      assert(position.hasContentDimensions && position.hasPixels);
-      if (position.pixels > position.minScrollExtent) {
+      if (!position.hasContentDimensions || !position.hasPixels) {
+        // It's possible that NestedScrollView built twice before layout phase
+        // in a same frame.
+        // This happens for example when FocusManager schedule a microTask
+        // that mark NestedScrollView dirty during `scheduleAttachRootWidget`.
```

Comment:
that markes ... during the warm up frame

Reply:
Thank you for your detailed grammar guide. : )